### ISSUE TITLE
Lift or explain the restriction on using unknown-offset timestamps in valid_values ranges

### ISSUE THREAD

**Issue by zslayton** [amzn/ion-schema#36]
The [`valid_values` section of the spec](https://amzn.github.io/ion-schema/docs/spec.html#valid_values) says:

> When a timestamp range is specified, neither end of the range may have an unknown local offset.

Additionally, [the current implementation](https://github.com/amzn/ion-schema-kotlin/blob/master/src/com/amazon/ionschema/internal/util/RangeIonTimestamp.kt#L61) prevents any `Timestamp` with a `localOffset` of `null` from being compared by throwing an exception:

```kt
    override fun contains(value: IonTimestamp): Boolean {
        // ValidValues performs this same check and adds a Violation
        // instead of invoking this method;  this if is here purely
        // as a defensive safety check, and will ideally never be true
        if (value.localOffset == null) {
            throw IllegalArgumentException("Unable to compare timestamp with unknown local offset: $value")
        }
        return delegate.contains(value.decimalMillis)
    }
```

The [`timestamp` section of the Ion spec](https://amzn.github.io/ion-docs/docs/spec.html#timestamp) says:

> Ion follows the “Unknown Local Offset Convention” of RFC3339:
>> If the time in UTC is known, but the offset to local time is unknown, this can be represented with an offset of “-00:00”. This differs semantically from an offset of “Z” or “+00:00”, which imply that UTC is the preferred reference point for the specified time. RFC2822 describes a similar convention for email.

Timestamps with an unknown offset are UTC timestamps--they represent a known instant. There's no reason we couldn't support comparisons between Timestamps with known/unknown offsets. We should either lift the restriction or explain why it's there.

**Comment by popematt** [comment]
I think there are two different problems here.

> The [valid_values section of the spec](https://amzn.github.io/ion-schema/docs/spec.html#valid_values) says:
> When a timestamp range is specified, neither end of the range may have an unknown local offset.

The first is this, the restriction on having timestamps with unknown offsets at the end of the range. This is strange and unjustified, but mostly harmless. It's a problem with the spec, and it can be fixed in Ion Schema 2.0.

> Additionally, [the current implementation](https://github.com/amzn/ion-schema-kotlin/blob/master/src/com/amazon/ionschema/internal/util/RangeIonTimestamp.kt#L61) prevents any Timestamp with a localOffset of null from being compared by throwing an exception

This problem is a buggy implementation. The spec does _not_ say that a timestamp with unknown offset cannot be compared to a range, and by not allowing unknown-offset timestamps to be compared to a range, it's excluding all timestamps with a precision of year, month, or day. (Some common use cases for low-precision timestamps include credit card expiry dates or birthdays, so this isn't a niche problem.)

I've created https://github.com/amzn/ion-schema-kotlin/issues/205 for tracking the bug in the implementation, and we can continue to use this issue to track the problem with the spec.

**Comment by popematt** [comment]
This issue has been resolved in Ion Schema 2.0.

There is no real justification for this restriction in Ion Schema 1.0, but we cannot remove the restriction now as that would be a potentially breaking change to the Ion Schema 1.0 specification.
